.github/workflows/publish-commit-bottles.yml

```diff
@@ -103,15 +105,23 @@ jobs:
             exit 1
           fi
 
-          echo "branch=$branch" >> "$GITHUB_OUTPUT"
+          if [ "$branch" = "master" ]
+          then
+            echo "branch=$head_repo_owner/$branch" >> "$GITHUB_OUTPUT"
+          else
+            echo "branch=$branch" >> "$GITHUB_OUTPUT"
+          fi
+          echo "origin_branch=$branch" >> "$GITHUB_OUTPUT"
           echo "remote=$remote" >> "$GITHUB_OUTPUT"
 
-          if "$pushable" && [ "$fork_type" != "Organization" ]
+          if "$pushable"
           then
-            exit 0
-          elif "$pushable"
-          then
-            MESSAGE="$ORG_FORK_MESSAGE"
+            if [ "$fork_type" != "Organization" ] || [ "$head_repo" = '${{github.repository}}' ]
+            then
+              exit 0
+            else
+              MESSAGE="$ORG_FORK_MESSAGE"
+            fi
           else
             MESSAGE="$NON_PUSHABLE_MESSAGE"
           fi
```

Comment:
This fix still didn't work on #127422, because:

```
$ gh api \
  --header 'Accept: application/vnd.github+json' \
  --header 'X-GitHub-Api-Version: 2022-11-28' \
  'repos/Homebrew/homebrew-core/pulls/127422' |
  jq .maintainer_can_modify
false
```